tests/drivers/dma/chan_blen_transfer/boards/nucleo_h743zi.conf

```diff
@@ -1,3 +1,16 @@
-CONFIG_DMA_TRANSFER_CHANNEL_NR_0=6
-CONFIG_DMA_TRANSFER_CHANNEL_NR_1=10
+CONFIG_DMA_TRANSFER_CHANNEL_NR_0=4
+CONFIG_DMA_TRANSFER_CHANNEL_NR_1=6
+CONFIG_DMA_LOOP_TRANSFER_NUMBER_OF_DMAS=2
+
+# Required by BDMA which only has access to
+# a cached SRAM4 section. All other DMAs also
+# has access to this section.
 CONFIG_NOCACHE_MEMORY=y
+CONFIG_DMA_LOOP_TRANSFER_SRAM_SECTION=".sram4"
+
+# Because the memory is not placed in the default
+# ".nocache" section the cache needs to be
+# invalidated manually
+CONFIG_CACHE=y
+CONFIG_CACHE_MANAGEMENT=y
+CONFIG_ARCH_CACHE=y
```

Comment:
You should be able to declare sram4 as `RAM_NOCACHE` using `zephyr,memory-region-mpu` property.
This would simplify handling of buffers on user side.

Question is to know if this is ok to declare sram4 as `RAM_NOCACHE` directly in .dtsi (and affect all current potential users) or should it be done as a configuration (either in .dts or .overlay) and a RAM configuration check in the driver. The former would be easier but the later offer more flexibility to users and is backward compatible (so has my preference).

Reply:
Oh, didn't know I could do that. That would be much easier than the manual cache handling. Will implement it early next week. And I agree, I would rather have it done as a configuration.

> and a RAM configuration check in the driver.

Do you mean that the driver should fail initialization if `nocache` isn't set of the `sram4` section? That would definitely make it much more user friendly, because cache issues are hard to spot and fix.

Reply:
> Do you mean that the driver should fail initialization if nocache isn't set of the sram4 section?

Yep

Reply:
Okay, I've changed the tests to use `zephyr,memory-region-mpu` and it's indeed much cleaner. 

However, I don't think it's currently possible (or easily implementable) to check if a given address is in a noncachable memory region. Would we consider adding this check required/blocking for now? There's nothing of the sort in the existing DMA drivers, which expects the user to handle all cache maintainence themselves.

Reply:
What about checking if dt property is set?

Reply:
> What about checking if dt property is set?

I thought about that, but it doesn't cover all the possible scenarios. Most unit tests for example use the section `.nocache` which is only defined in the linker, and has no reference (that I know of) in the DTS.

Reply:
> Most unit tests for example use the section .nocache which is only defined in the linker, and has no reference (that I know of) in the DTS.

My understanding is that BDMA (in current implementation) can only work with SRAM4, which we expect to be defined with RAM_NOCACHE attribute. We are able to check that this specific condition is set, right ?

Reply:
Check added. I ended up needing a `strcmp`, but seeing it's a once-off during init I think it would be okay.